gpbft/emulator/assertions.go

```diff
@@ -0,0 +1,131 @@
+	r.Equal(gpbft.PREPARE_PHASE, subject.Vote.Step)
+	r.Equal(round, subject.Vote.Round)
+	r.Equal(value, subject.Vote.Value)
+	r.Equal(instance.id, subject.Vote.Instance)
+	r.Equal(instance.supplementalData, subject.Vote.SupplementalData)
+	r.Equal(justification, subject.Justification)
+	r.Empty(subject.Ticket)
+}
+
+// CommitWithJustification asserts that the given subject message belongs to the
+// given instance, round and votes for the given vote with the expected
+// justification.
+func (r *Assertions) CommitWithJustification(subject *gpbft.GMessage, instance *Instance, round uint64, vote gpbft.ECChain, justification *gpbft.Justification) {
+	r.Equal(gpbft.COMMIT_PHASE, subject.Vote.Step)
+	r.Equal(round, subject.Vote.Round)
+	r.Equal(instance.supplementalData, subject.Vote.SupplementalData)
+	r.Equal(instance.id, subject.Vote.Instance)
+	r.Equal(vote, subject.Vote.Value)
+	r.Equal(justification, subject.Justification)
+	r.Empty(subject.Ticket)
+}
+
+// CommitForBottom asserts that the given subject message belongs to the given
+// instance, round and is a COMMIT for bottom.
+func (r *Assertions) CommitForBottom(subject *gpbft.GMessage, instance *Instance, round uint64) {
+	r.Equal(gpbft.COMMIT_PHASE, subject.Vote.Step)
+	r.Equal(round, subject.Vote.Round)
+	r.Equal(instance.supplementalData, subject.Vote.SupplementalData)
+	r.Equal(instance.id, subject.Vote.Instance)
+	r.True(subject.Vote.Value.IsZero())
+	r.Nil(subject.Justification)
+	r.Empty(subject.Ticket)
+}
+
+// Converge asserts that the given subject message belongs to the given instance,
+// converges on the given vote and has the expected justification.
+func (r *Assertions) Converge(subject *gpbft.GMessage, instance *Instance, round uint64, vote gpbft.ECChain, justification *gpbft.Justification) {
+	r.Equal(gpbft.CONVERGE_PHASE, subject.Vote.Step)
+	r.Equal(round, subject.Vote.Round)
+	r.Equal(vote, subject.Vote.Value)
+	r.Equal(instance.id, subject.Vote.Instance)
+	r.Equal(instance.supplementalData, subject.Vote.SupplementalData)
+	r.Equal(justification, subject.Justification)
+	r.NotEmpty(subject.Ticket)
+}
+
+// Decide asserts that the given subject message belongs to the given instance,
+// decides the given vote and has the expected justification.
+func (r *Assertions) Decide(subject *gpbft.GMessage, instance *Instance, vote gpbft.ECChain, justification *gpbft.Justification) {
+	r.Equal(gpbft.DECIDE_PHASE, subject.Vote.Step)
+	r.Zero(subject.Vote.Round)
+	r.Equal(vote, subject.Vote.Value)
+	r.Equal(instance.id, subject.Vote.Instance)
+	r.Equal(instance.supplementalData, subject.Vote.SupplementalData)
+	r.Equal(justification, subject.Justification)
+	r.Empty(subject.Ticket)
+}
+
+// Consensus asserts that the given instance has reached consensus for the expected chain.
+func (r *Assertions) Consensus(instance *Instance, expect gpbft.ECChain) {
```

Comment:
"Consensus" isn't the right term for a single instance. I suggest `Decision` or `Decided`.